tests/hmm/hmumu_utils.py

```diff
@@ -1376,6 +1330,64 @@ def get_selected_muons(
         "additional_muon_sel": additional_muon_sel,
     }
 
+#Corrects the muon momentum and isolation, if a matched FSR photon with dR<0.4 is found
+@numba.njit
+def correct_muon_with_fsr(
+        muons_offsets, fsr_offsets,
+        muons_pt, muons_eta, muons_phi, muons_mass, muons_iso, muons_fsrIndex,
+        fsr_pt, fsr_eta, fsr_phi
+    ):
+
+    for iev in numba.prange(len(muons_offsets) - 1):
+
+        #loop over muons in event
+        mu_first = muons_offsets[iev]
+        mu_last = muons_offsets[iev + 1]
+        for imu in range(mu_first, mu_last):
+            #relative FSR index in the event
+            fsr_idx_relative = muons_fsrIndex[imu]
+
+            if (fsr_idx_relative >= 0) and (muons_pt[imu]>20):
+                #absolute index in the full FSR vector for all events
+                ifsr = fsr_offsets[iev] + fsr_idx_relative
+                mu_kin = {"pt": muons_pt[imu], "eta": muons_eta[imu], "phi": muons_phi[imu], "mass": muons_mass[imu]}
+                fsr_kin = {"pt": fsr_pt[ifsr], "eta": fsr_eta[ifsr], "phi": fsr_phi[ifsr],"mass": 0}
+
+                # dR between muon and photon
+                deta = muons_eta[imu] - fsr_eta[ifsr]
+                dphi = deltaphi_cpu_devfunc(muons_phi[imu], fsr_phi[ifsr])
+                dr = NUMPY_LIB.sqrt(deta**2 + dphi**2)
+
+                update_iso = dr<0.4
+
+                if update_iso:
+                    muons_iso_abs = (muons_iso[imu]*muons_pt[imu] - fsr_pt[ifsr])
+
+                #compute and set corrected momentum
+                px_total = 0
+                py_total = 0
+                pz_total = 0
+                e_total = 0
+                for obj in [mu_kin, fsr_kin]:
+                    px = obj["pt"] * np.cos(obj["phi"])
+                    py = obj["pt"] * np.sin(obj["phi"])
+                    pz = obj["pt"] * np.sinh(obj["eta"])
+                    e = NUMPY_LIB.sqrt(px**2 + py**2 + pz**2 + obj["mass"]**2)
+                    px_total += px
+                    py_total += py
+                    pz_total += pz
+                    e_total += e
```

Comment:
I don't see this e_total being used anywhere else. If it doesn't affect the isolation calculation, perhaps we don't need to have extra lines? Otherwise, everything else looks good!